pkg/kademlia/dialer_test.go

```diff
@@ -177,7 +177,7 @@ func TestSlowDialerHasTimeout(t *testing.T) {
 	// TODO: also use satellites
 	peers := planet.StorageNodes
 
-	{ // PingNode
+	func() { // PingNode
```

Comment:
why do you wrap these in `func`?

Reply:
So that the defer's inside execute immediately rather than all at once at the end of the test.

Reply:
ah yes makes sense. good one